src/ophyd_async/core/signal.py

```diff
@@ -253,6 +257,52 @@ def set_sim_callback(signal: Signal[T], callback: ReadingValueCallback[T]) -> No
     return _sim_backends[signal].set_callback(callback)
 
 
+async def verify_readable(
+    func: Callable[[], dict[str, Any] | Awaitable[dict[str, Any]]],
+    expectation: Dict[str, Any],
+) -> None:
+    """Loop thorough dictionary and verify it against
+    a expected dictionary (readable)"""
+    if asyncio.iscoroutinefunction(func):
+        result = await func()
+    else:
+        result = func()
+
+    for signal in expectation:
+        for field in expectation[signal]:
+            if field == "timestamp":
+                assert isinstance(result[signal]["timestamp"], float)
+            else:
+                assert result[signal][field] == expectation[signal][field]
```

Comment:
I think it's fine to just do 

```suggestion
    assert result == expectation
```

And expect people to do things like

```python
from unittest.mock import ANY

await assert_reading(my_detector, {"value": 0.1, "timestamp": ANY})
```

It also gives them more power to make other attributes `ANY` if they don't care about the values